sms-authenticator/src/main/java/netzbegruenung/keycloak/authenticator/SmsAuthenticator.java

```diff
@@ -62,11 +65,11 @@ public void authenticate(AuthenticationFlowContext context) {
 		RealmModel realm = context.getRealm();
 
 		Optional<CredentialModel> model = context.getUser().credentialManager().getStoredCredentialsByTypeStream(SmsAuthCredentialModel.TYPE).findFirst();
-		String mobileNumber = "";
+		String mobileNumber;
 		try {
-			mobileNumber = JsonSerialization.readValue(model.get().getCredentialData(), SmsAuthCredentialData.class).getMobileNumber();
+			mobileNumber = JsonSerialization.readValue(model.orElseThrow().getCredentialData(), SmsAuthCredentialData.class).getMobileNumber();
```

Comment:
Potential NullPointerException here, replaced with "orElseThrow()" to be explicit about it.